pjsip/src/pjsip/sip_msg.c

```diff
@@ -898,8 +898,7 @@ static int pjsip_generic_int_hdr_print( pjsip_generic_int_hdr *hdr,
 static pjsip_generic_int_hdr* pjsip_generic_int_hdr_clone( pj_pool_t *pool, 
                                                    const pjsip_generic_int_hdr *rhs)
 {
-    pjsip_generic_int_hdr *hdr = PJ_POOL_ALLOC_T(pool, pjsip_generic_int_hdr);
-    pj_memcpy(hdr, rhs, sizeof(*hdr));
+    pjsip_generic_int_hdr *hdr = pjsip_generic_int_hdr_create(pool, &rhs->name, rhs->ivalue);
```

Comment:
Just realized that it is a generic header, using `pjsip_generic_int_hdr_create()` will always reset the header type to `PJSIP_H_OTHER`. So from the original code, only adding header name duplication (especially if the type is `PJSIP_H_OTHER`) is sufficient?